myosuite/envs/myo/myochallenge/run_track_v0.py

```diff
@@ -0,0 +1,677 @@
+        self._setup(**kwargs)
+
+    def _setup(self,
+               obs_keys: list = DEFAULT_OBS_KEYS,
+               weighted_reward_keys: dict = DEFAULT_RWD_KEYS_AND_WEIGHTS,
+               reset_type='init',
+               terrain='FLAT',
+               hills_difficulties=(0,0),
+               rough_difficulties=(0,0),
+               stairs_difficulties=(0,0),
+               real_length=15,
+               real_width=1,
+               distance_thr = 10,
+               init_pose_path=None,
+               **kwargs,
+               ):
+
+        self.startFlag = False
+        # Terrain type
+        self.terrain_type = TerrainTypes.FLAT
+
+        # Env initialization with data
+        if init_pose_path is not None:
+            file_path = os.path.join(init_pose_path)
+            self.INIT_DATA = np.loadtxt(file_path, skiprows=1, delimiter=',')
+            self.init_lookup = self.generate_init_lookup(keys=np.arange(48), value='e_swing')
+            self.init_lookup = self.generate_init_lookup(keys=np.arange(48, 99), value='l_swing', existing_dict=self.init_lookup)
+            self.init_lookup = self.generate_init_lookup(keys=np.arange(99, 183), value='e_stance', existing_dict=self.init_lookup)
+            self.init_lookup = self.generate_init_lookup(keys=np.arange(183, 247), value='l_stance', existing_dict=self.init_lookup)
+            with open(file_path) as csv_file:
+                csv_reader = csv.DictReader(csv_file)
+                temp = dict(list(csv_reader)[0])
+                headers = list(temp.keys())
+            self.imitation_lookup = dict(zip(headers, range(len(headers))))
+
+        # OSL specific init
+        self.OSL_CTRL = MyoOSLStateMachine(np.sum(self.sim.model.body_mass), 'e_stance')
+
+        self.muscle_space = self.sim.model.na # muscles only
+        self.full_ctrl_space = self.sim.model.nu # Muscles + actuators
+        self._get_actuator_params()
+
+        self._setup_convenience_vars()
+        self.distance_thr = distance_thr
+        self.trackfield = TrackField(
+            sim=self.sim,
+            rng=self.np_random,
+            rough_difficulties=rough_difficulties,
+            hills_difficulties=hills_difficulties,
+            stairs_difficulties=stairs_difficulties,
+        )
+        self.real_width = real_width
+        self.real_length = real_length
+        self.reset_type = reset_type
+        self.terrain = terrain
+        self.grf_sensor_names = ['l_foot', 'l_toes']
+        super()._setup(obs_keys=obs_keys,
+                       weighted_reward_keys=weighted_reward_keys,
+                       reset_type=reset_type,
+                       **kwargs
+                       )
+        self.init_qpos[:] = self.sim.model.keyframe('osl_forward').qpos.copy()
+        self.init_qvel[:] = 0.0
+        self.assert_settings()
+        self.startFlag = True
+
+    def assert_settings(self):
```

Comment:
What's this?

Reply:
was needed but is no moved outside the setup

Reply:
and assert_settings was deleted.